jraft-core/src/main/java/com/alipay/sofa/jraft/storage/snapshot/local/LocalSnapshotWriter.java

```diff
@@ -95,21 +97,32 @@ public void close() throws IOException {
 
     @Override
     public void close(final boolean keepDataOnError) throws IOException {
+        checkState();
         this.snapshotStorage.close(this, keepDataOnError);
+        this.closed = true;
```

Comment:
Maybe use CAS for `this.close` since `this.snapshotStorage.close` might be called concurrently.

Reply:
It notes that itself is not thread-safe, I think this checking is enough.